app/middlewares/user.js

```diff
@@ -0,0 +1,65 @@
+const { body, check, validationResult } = require('express-validator/check'),
+  User = require('../models').User,
+  logger = require('../logger'),
+  errors = require('../errors');
+
+const emailBelongsToWolox = email => {
+  // checks if email is valid and if it's domain belongs to wolox.
+  // regex exp replaces all characters up-to and including the last '@' with empty: ''
+  // so: some@email@wolox.com.ar => wolox.com.ar
+  return email.replace(/.*@/, '') === 'wolox.com.ar';
+};
+
+const emailIsRegistered = email => {
+  return User.count({ where: { email } }).catch(e => {
+    throw errors.databaseError(e.message);
+  });
+};
+
+exports.validateEmail = body('email')
+  .isEmail()
+  .withMessage('The email is required')
+  .normalizeEmail()
+  .custom(email => emailBelongsToWolox(email))
+  .withMessage('Email is required and must belong to wolox');
+
+const validateTextField = field =>
+  body(field)
+    .exists()
+    .withMessage(`The ${field} field is required`)
+    .isString()
+    .withMessage(`The ${field} field must be string`)
+    .not()
+    .isEmpty()
+    .withMessage(`The ${field} field must not be empty`);
+
+exports.validateFirstName = validateTextField('firstName');
+exports.validateLastName = validateTextField('lastName');
+
+exports.validatePassword = body('password')
+  .isLength({ min: 8 })
+  .withMessage('Password must be at least 8 characters long')
+  .matches(/\d/)
+  .withMessage('Password must be alphanumeric');
+
+exports.validateEmailIsNotRegistered = body('email').custom(email => {
+  return emailIsRegistered(email).then(isRegistered => {
```

Comment:
This check move to controller. The middleware only validate if the keys exist and are corrects